Model/BalancepayMethod.php

```diff
@@ -403,14 +403,19 @@ public function capture(InfoInterface $payment, $amount)
                 }
             }
 
-            $this->requestFactory
+            $response = $this->requestFactory
                 ->create(RequestFactory::CAPTURE_REQUEST_METHOD)
                 ->setPayment($payment)
                 ->setAmount($amount)
                 ->setBalanceVendorId($balanceVendorId)
                 ->process();
-        }
 
+            $charges = $response['charges'];
+            if (is_array($charges) && isset($charges[0])) {
```

Comment:
what is `$payment`? is it saved on the DB?
if so, is there a relation between `payment` and `invoice`?